File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.py
def f(number):
    bin_num = bin(number)[2:]

    if '0' not in bin_num:
        return int('10' + bin_num[1:], 2)

    bin_num = list(bin_num)
    for i in range(len(bin_num)):
        if bin_num[-i-1] == '0':
            bin_num[-i-1] = '1'
            break

    if i > 0:
        bin_num[-i] = '0'

    return int(''.join(bin_num), 2)


def solution(numbers):
    answer = [f(number) for number in numbers]
    return answer

File: 프로그래머스/2/77885. 2개 이하로 다른 비트/2개 이하로 다른 비트.js
function solution(numbers) {
    return numbers.map(v => {
        let n = '0'+v.toString(2);
        if (n[n.length - 1] === '0') {
            n = n.slice(0, n.length-1) + '1';
        } else {
            const i = n.lastIndexOf('01');
            n = n.slice(0,i) + '10' + n.slice(i+2, n.length);
        }
        
        return parseInt(n, 2);
    })
}